### ISSUE TITLE
[FEATURE REQUEST]: podman machine debugging helper

### ISSUE THREAD

**Issue by cevich** [containers/podman#14684]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

Podman machine's VM is very difficult to debug if there are boot and/or ignition-script problems.  The `podman machine init` command should accept an optional argument: Path to a file where the VM's serial port output is **appended** upon `start` and closed upon `stop`.

**Steps to reproduce the issue:**

0. `/path/to/file/` contains pre-existing/valuable debugging info.

1. `podman machine init --serial_output=/path/to/file foobar`

2. `podman machine start`

3. `/path/to/file` is updated with serial port output from VM `foobar`.

4. `podman machine stop foobar`

5. `/path/to/file` is no longer updated updated with serial port output from VM `foobar`.

**Describe the results you received:**

**Describe the results you expected:**

When CI testing podman machine changes, if there there is an ignition or boot problem, no state is preserved for debugging.

**Additional information you deem important (e.g. issue happens only occasionally):**

CI will happily gather an output file containing serial port output for later examination if ever required.

**Output of `podman version`:**

```
(paste your output here)
```

**Output of `podman info --debug`:**

```
(paste your output here)
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(paste your output here)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes/No

**Additional environment details (AWS, VirtualBox, physical, etc.):**

I specified *append* for the serial output file behavior, since a CI test may start/stop a VM multiple times and a hang/crash may not occur only on the initial startup.  This is a "nice to have" but not a hard requirement if implementation proves too difficult.

**Comment by jwhonce** [comment]
Rather than a new command, would it be easier to do this additional logging on `--log-level=trace` ?

**Comment by cevich** [comment]
Since the VM image is downloaded dynamically, it's extra hard to predict when it will fail to boot/kernel panic/oom/etc.  I had in mind we'd just make all the podman-machine tests always log this, all the time.  However it's done, the point was that sticking it into a file instead of stdout/stderr will make both collecting and looking at it easier on the eyes.  So, I was hoping something could just be stuck onto the QEMU, like (thanks stack overflow): `-chardev stdio,id=char0,logfile=serial.log,signal=off -serial chardev:char0` (not tested).  In any case I understand that logging from an asynchronous process/service like this can be tricky on the best of days.  I'll leave it to y'all to come up with the most sensible implementation.

**Comment by cevich** [comment]
I think @baude was working on adding this?

**Comment by matthenry87** [comment]
I am unable to SSH into my applehv machine running on my M2 MacBook ARM laptop.

```bash
❯ podman machine ssh --log-level=debug
INFO[0000] podman filtering at log level debug
DEBU[0000] Using Podman machine with `applehv` virtualization provider
DEBU[0000] socket length for /Users/REDACTED/.config/containers/podman/machine/applehv is 57
DEBU[0000] socket length for /Users/REDACTED/.local/share/containers/podman/machine/applehv is 62
DEBU[0000] socket length for /Users/REDACTED/.local/share/containers/podman/machine/applehv/cache is 68
DEBU[0000] socket length for /var/folders/9y/_f3s88y942g2nq1vhpks0f4m0000gp/T/podman is 55
DEBU[0000] socket length for /Users/REDACTED/.config/containers/podman/machine/applehv/podman-machine-default.json is 85
Connecting to vm podman-machine-default. To close connection, use `~.` or `exit`
DEBU[0000] Executing: ssh [-i /Users/REDACTED/.local/share/containers/podman/machine/machine -p 62892 root@localhost -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o LogLevel=ERROR -o SetEnv=LC_ALL= -t]
DEBU[0000] Called machine ssh.PersistentPostRunE(podman machine ssh --log-level=debug)
DEBU[0000] Shutting down engines
```

Some insight on the machine side sure would be nice.

**Comment by danishprakash** [comment]
I'd like to take a shot at this, any clue on what's the status on this?

**Comment by cevich** [comment]
@danishprakash I don't believe anybody else is currently working on this.  If you're able to crack it, it would be of massive benefit to many people, earning you a big pile of :cookie: (cookies) and gratitude.
